packages/editor/src/components/post-card-panel/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Icon,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	PanelBody,
+} from '@wordpress/components';
+import { page as pageIcon, navigation, symbol } from '@wordpress/icons';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { humanTimeDiff } from '@wordpress/date';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+const CARD_ICONS = {
+	wp_block: symbol,
+	wp_navigation: navigation,
+};
+
+export default function PostCardPanel( { className, actions, children } ) {
+	const { modified, title, type, templateInfo } = useSelect( ( select ) => {
+		const {
+			getEditedPostAttribute,
+			getCurrentPostType,
+			getCurrentPostId,
+			__experimentalGetTemplateInfo,
+		} = select( editorStore );
+		const { getEditedEntityRecord } = select( coreStore );
+		const _type = getCurrentPostType();
+		const _id = getCurrentPostId();
+		let _templateInfo;
+		if ( _type === 'wp_template' || _type === 'wp_template_part' ) {
+			const _record = getEditedEntityRecord( 'postType', _type, _id );
+			_templateInfo = __experimentalGetTemplateInfo( _record );
+		}
+		return {
+			title: _templateInfo?.title || getEditedPostAttribute( 'title' ),
+			modified: getEditedPostAttribute( 'modified' ),
+			id: _id,
+			type: _type,
+			templateInfo: _templateInfo,
+		};
+	} );
+	const description =
+		type === 'wp_template' || type === 'wp_template_part'
+			? templateInfo?.description
+			: sprintf(
+					// translators: %s: Human-readable time difference, e.g. "2 days ago".
+					__( 'Last edited %s' ),
+					humanTimeDiff( modified )
+			  );
+	const icon = CARD_ICONS[ type ] || templateInfo?.icon || pageIcon;
```

Comment:
There's a lot of inconsistencies and conditions in the way we compute "title", "description" and "icon" between post types. Can we track this. I'd like to have a unique way to retrieve these informations regardless of the post type. (Also this logic seems duplicated in multiple places for instance the document bar)

Reply:
➕ to this. We should have a way to retrieve this for any post type. Additionally the `pageIcon` as default doesn't seem right, since we show this for post too and other post types. I'm wondering if we should fallback to the postType.icon.

Reply:
I added a new private selector getPostIcon, it should be able to be used in the cases where we need icons like the document bar, if we agree with this selector as a side PR I can propose one where the selector becomes widely used.